src/otel/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+use opentelemetry::global::BoxedSpan;
+use opentelemetry::trace::{
+    Span, SpanBuilder, SpanKind, TraceId, Tracer, TracerProvider,
+};
+use opentelemetry::{global, KeyValue};
+use opentelemetry_otlp::Protocol;
+use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::trace::{Config, BatchConfigBuilder};
+use opentelemetry_sdk::{runtime, Resource};
+use std::ffi::{CStr, CString};
+use std::ptr;
+use std::ptr::addr_of;
+use std::sync::{Arc, OnceLock};
+use std::time::Duration;
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+
+// Stored sender channel to send spans or a shutdown message to within the
+// Tokio runtime.
+unsafe fn span_tx(destruct: bool) -> *const OnceLock<Sender<SpanMessage>> {
```

Comment:
I wouldn't use bool as an argument, because when calling it you don't know what the argument means:

```
span_tx(true)
span_tx(false)
```

My suggestion is to have two consts like

```
const SEND_TX: bool = true;
const GET_TX: bool = false;
```
and then use those as arguments:

```
span_tx(SEND_TX);
span_tx(GET_TX);
```

Var names can be adjusted as needed.

Reply:
> `nxt_AREA_`

Sure, but I may tweak it a bit just so that its clear in GDB which symbols are coming from rust code.